FFCP updates. Always update XY software endstops on tool_change

mikeshub opened the pull request:
Thermal protection was not working with double nozzle printing.

Change current and kinematics to be more reasonable.

Fix and cleanup pins header file.

Added a text file with the setting to use with Cura.


### Description

This pull request makes no major changes. This is a small set of tweaks to improve double nozzle printing. Individual axis homing and not disabling Z stepper are used for manually leveling the bed.

### Benefits

This also fixes the pins header file to include the more than 70 pin definitions. 

### Related Issues

Tool change swap software endstops issue:

https://github.com/MarlinFirmware/Marlin/issues/14128

mikeshub committed:
Update FF CP pins, config, and add a doc describing use with Cura

Thermal protection was not working with double nozzle printing.

Change current and kinematics to be more reasonable.

Fix and cleanup pins header file.

mikeshub committed:
Machine should not do unknown raise

This can result in the bed being driven past its limit.

mikeshub committed:
Add PCA9533 init to LED code

LEDs were not working due to not being initialized.

mikeshub committed:
Put the LED init call in the right place

mikeshub committed:
Add caselight default to startup

mikeshub committed:
Fix software end stops on tool change

mikeshub committed:
Update Configuration.h

Fix X min software endstop position

thinkyhead committed:
Fix spacing, remove pins debug include

thinkyhead committed:
Update pins_MIGHTYBOARD_REVE.h

thinkyhead committed:
Update Configuration_adv.h

thinkyhead committed:
Update Configuration_adv.h

thinkyhead committed:
Update Configuration.h

thinkyhead commented on Marlin/src/module/tool_change.cpp:
```diff
@@ -750,8 +750,10 @@ void tool_change(const uint8_t tmp_extruder, const float fr_mm_s/*=0.0*/, bool n
+      #if HAS_SOFTWARE_ENDSTOPS
         update_software_endstops(X_AXIS, active_extruder, tmp_extruder);
+        update_software_endstops(Y_AXIS, active_extruder, tmp_extruder);
+        update_software_endstops(Z_AXIS, active_extruder, tmp_extruder);
```
Do you suppose this is okay in all cases?

thinkyhead replied:
Since Dual X ignores Y and Z and the compiler isn't liable to catch this, these lines won't be compiled in for Dual X.

mikeshub replied:
I think your ifdefs are the way to go.

thinkyhead committed:
For Dual X update_software_endstops ignores YZ

thinkyhead committed:
Merge remote-tracking branch 'upstream/bugfix-2.0.x' into bf2_fixup_delete_14125

mikeshub commented:
Yeah so thermal protection for the hotends has to be disabled otherwise the temp changes commanded by Cura for double nozzle work cause the machine to shut down mid print. The bed could be left enabled, but I was unable to find settings that would work for the hotends.

mikeshub committed:
Fix calls to update software endstops

This should allow it to compile under all targets. This is just a bit messy. Maybe there is some way we could clean up section? Maybe get rid of the #if on motion.cpp line 466 and motion.h line 126? I don't think old and new tool index will cause problems . I'll mess with that a little. For now this should fix the build failure issue.

mikeshub committed:
Revert "Fix calls to update software endstops"

This reverts commit 4486924f429be6ce455115dba2fa4c89b121d401.

mikeshub committed:
Merge branch 'bugfix-2.0.x' of https://github.com/mikeshub/Marlin into bugfix-2.0.x

mikeshub committed:
Update tool change code endstop update handling

This should fix the compilation errors that come from updating the endstops when hotend offsets aren't defined.

mikeshub commented:
I updated the code in tool_change.cpp to have an #if in the same way that the call to update_software_endstops does. This should fix travis-ci build errors.

mikeshub commented:
This stuff is all squared away. Any chance we could get this merged and closed out? The LED stuff still needs to be added. Thanks!

thinkyhead committed:
Update CuraSettings.txt

thinkyhead committed:
Update Configuration_adv.h

thinkyhead committed:
Update tool_change.cpp

thinkyhead commented on Marlin/src/module/tool_change.cpp:
```diff
@@ -750,8 +750,17 @@ void tool_change(const uint8_t tmp_extruder, const float fr_mm_s/*=0.0*/, bool n
+        update_software_endstops(X_AXIS _EXT_ARGS);
+        #if DISABLED(DUAL_X_CARRIAGE)
+          update_software_endstops(X_AXIS _EXT_ARGS);
+          update_software_endstops(X_AXIS _EXT_ARGS);
```
@mikeshub — I'm going to assume you meant Y and Z here.

thinkyhead committed:
Update tool_change.cpp

mikeshub reviewed (commented):
Yeah I think so. My power is out and will be for a couple more days so I'll have to wait until then to check.
